src/js/model/macro/macro.js

```diff
@@ -0,0 +1,60 @@
+var AMPParse = AMPParse || {};
+
+AMPParse.buildMacro = function(hexadecimal) {
+
+    if (!(hexadecimal instanceof AMPHexConsumer)) {
+        throw new TypeError("Did not receive an AMPHexConsumer");
+    }
+
+    var nibblesConsumed = 0;
+    var tempValue;
+    var returnValue = {"type": "Managed Identifier"};
+
+    try {
+        tempValue = AMPParse.buildManagedIdentifier(hexadecimal);
+        nibblesConsumed += tempValue.nibblesConsumed;
+        returnValue.identifier = tempValue.returnValue;
+    } catch(err) {
+        err.message = "Failed to get the identifier for a macro: " + err.message;
+        throw err;
+    }
+
+    if (returnValue.identifier.header.isParametrized) {
+        var err = new Error("A macro identifier cannot contain parameterized OID");
+        err.nibblesConsumed += nibblesConsumed;
+        throw err;
+    }
+
+    /*
+    TODO: how do we know if the macro is defined outside the ADM
+    var header = returnValue.identifier.header;
+    if (returnValue.identifier.value.isExternal) {
+        if (header.hasIssuer === false) {
+            var err = new Error("An externally defined Macro must contain an issuer");
+            err.nibblesConsumed += nibblesConsumed;
+            throw err;
+        }
+    } else {
+        if (header.hasIssuer || header.hasTag) {
+            var err = new Error("A macro that is defined in an ADM must not have an" +
+                " issuer field and must not have a tag field.")
+            err.nibblesConsumed += nibblesConsumed;
+            throw err;
+        }
+    }*/
```

Comment:
You can determine if it's outside the ADM if an OID's extraInfo property is "Unknown OID" as you can see [here](https://github.com/emanguy/amp-hex-translator/blob/master/src/js/model/oid/oid.js#L92).